### PR TITLE
Avoid state mutation in reducers/addons.js

### PR DESCRIPTION
**Pull request by arku**
Fixes https://github.com/mozilla/addons-frontend/issues/6147

This patch removes the usage of state mutation pattern in `reducers/addons.js`

### CODE REVIEW

File: src/core/reducers/addons.js [hunk 1]
```diff
@@ -385,31 +385,45 @@ export default function addonsReducer(
...
-
-        if (addon.guid) {
-          // `guid` is already "normalized" with the `getGuid()` function in
-          // `createInternalAddon()`.
-          byGUID[addon.guid] = addon.id;
-        }
-      });
-
-      return {
+        // $FLOW_FIXME
```
Comment (arku) [inline src/core/reducers/addons.js]: @kumar303 I am not sure how to keep Flow happy with the usage of the object spread operator over strings - hence `$FLOW_FIXME`. How can we fix this?

File: src/core/reducers/addons.js [hunk 2]
```diff
@@ -385,31 +385,48 @@ export default function addonsReducer(
...
-        ...state,
-        byID,
-        byGUID,
-        bySlug,
-        loadingBySlug,
-      };
+      return loadedAddons.reduce(
+        (acc, loadedAddon) => {
+          const addon = createInternalAddon(loadedAddon);
+          // $FLOW_FIXME
```
Comment (arku) [inline src/core/reducers/addons.js]: @kumar303 I am not sure how to keep Flow happy with the usage of the object spread operator over strings - hence $FLOW_FIXME. How can we fix this?

**Comment by kumar303** [comment]
Hi @arku. Thanks for taking the time to work on a patch! I'm going to close this because https://github.com/mozilla/addons-frontend/issues/6147 is not in a good enough state for a contributor to work on right now.

Please take a look at any issue labelled as a [good first bug](https://github.com/mozilla/addons-frontend/labels/contrib%3A%20good%20first%20bug) or one labelled as [contrib: welcome](https://github.com/mozilla/addons-frontend/labels/contrib%3A%20welcome). Thanks for your understanding. Let me know if you have questions about any of those issues.

**Comment by arku** [comment]
@kumar303 No worries!

**Pull request closed without merging**